modules/shared/src/main/scala/org/tesselation/kryo/KryoSerializer.scala

```diff
@@ -31,18 +34,23 @@ object KryoSerializer {
       }
     }(_ => Applicative[F].unit)
 
-  def forAsync[F[_]: Async](registrar: Map[Class[_], Int]): Resource[F, KryoSerializer[F]] = make[F](registrar).map {
-    kryoPool =>
-      new KryoSerializer[F] {
-        def serialize(anyRef: AnyRef): Either[Throwable, Array[Byte]] =
-          Either.catchNonFatal {
-            kryoPool.toBytesWithClass(anyRef)
-          }
-
-        def deserialize[T](bytes: Array[Byte]): Either[Throwable, T] =
-          Either.catchNonFatal {
-            kryoPool.fromBytes(bytes).asInstanceOf[T]
-          }
-      }
+  def forAsync[F[_]: Async](
+    registrar: Map[Class[_], Int],
+    migrations: List[Migration[AnyRef, AnyRef]] = List.empty
+  ): Resource[F, KryoSerializer[F]] = make[F](registrar).map { kryoPool =>
+    val migrationsMap = migrations.map(_.toPair).toMap
```

Comment:
it's used once only, so you can use it there explicitly:
```
migrations
  .map(_.toPair)
  .toMap
  .getOrElse...
```

Reply:
I think it might recalculate the map every time we deserialize something. I think that compiler isn't that smart to cut it off after `.toMap`

Reply:
Both ways it will recalculate IMO

Reply:
I checked by adding stdout logs:
```
.map { m =>
  val p = m.toPair
  println(s"mapping ${p._1}")
  p
}.toMap
```
and it recalculates the whole map during each deserialization. If we keep it as a variable outside of the `deserialize` function (as it is), it doesn't recalculate it.

Reply:
Yep, you're right. I forget we're using a Resource here that is initialized once only